doc/tutorials/enable_software_sram.rst

```diff
@@ -0,0 +1,129 @@
+
+3. CRL initializes software SRAM by setting up cache ways and mapping locked cache to memory space.
+
+4. ACRN hypervisor:
+	- Reserve the address range for software SRAM and remap software SRAM region to guest address space.
+	- Build and inject virtual RTCT(vRTCT) to RTVM as an ACPI table.
+	- Setup protection for software SRAM from being flushed runtimely, such as trap and emulate `wbinvd` instruction from guest.
+
+5. Real-Time Configuration Driver (RTVM Kernel) initializes when valid RTCT is detected and creates device node `/dev/tcc_buffer`.
+
+6. User space applications or libraries query configurations of software SRAM and request software SRAM buffers by issuing command to `/dev/tcc_buffer`.
+
+7. User space applications or libraries map software SRAM bufer to user space by issuing command to `/dev/tcc_buffer`.
+
+8. User space applications can access the low-latency software SRAM buffers.
+
+Design assumptions
+==================
+
+1. One RTVM at most, either pre-launch RTVM or pre-launch RTVM, can be supported for specific ACRN configuration. Will update this design when ACRN supports multiple RTVMs.
+
+2. Software SRAM and CAT features can't co-exist for current design, only one of them is allowed for one ACRN configuration.
+
+3. So far, software SRAM can be supported on Intel® TGL and EHL platforms only.
+
+4. When software SRAM is enabled for pre-launch RTVM, ACRN pass-through all software SRAM resource to this RTVM， which means:
+   - Platform RTCT instead of vRTCT will be injected to RTVM.
+   - RTVM access software SRAM region by host physical address(HPA), or say, identical mapping is used for case.
+
+5. When trying to enable software SRAM for pre-launch RTVM, L2 cache software SRAM is NOT supported， or say, ONLY L3 cache can be configured in platform RTCT.
+
+6. Please follow the latest Intel® TCC Tools guide, such as `Get Started with Intel® TCC Tools 2021.1`_ to confgiure software SRAM for specific platform:
+
+.. _Get Started with Intel® TCC Tools 2021.1:
+   https://software.intel.com/content/www/us/en/secure/develop/documentation/get-started-with-intel-time-coordinated-computing-tools-2021-1/top.html
+
+Enable Software SRAM for Pre-launch RTVM
+****************************************
+
+Please follow below steps to enable software SRAM support for pre-launch RTVM:
+
+1. Follow Intel® TCC Tools guide to configure software SRAM on target platform, according to previous `Design assumptions`, only L3 cache can be configured. the native RTCT ACPI table can be found at `/sys/firmware/acpi/tables/PTCT` when it is configured successfully.
+
+2. Configure ACRN:
+
+  - Stitch native RTCT ACPI table to `VM0_ACPI.bin` binary, which will be parsed to pre-launch RTVM as a multiboot module.
+
+  - Pre-launch RTVM shall be enabled for current ACRN configuration.
+
+  - Set `PSRAM_ENABLED` to `y` for pre-launch RTVM in ACRN configure file(`hybrid_rt.xml`), path to this file is platform specific.
```

Comment:
We shouldn't edit XML files directly, let's redirect users to the ACRN configuration tool instead.

Reply:
make sense, will update this, thank you @gvancuts